templates/build-dockerfiles/manifests.yaml

```diff
@@ -0,0 +1,214 @@
+            arguments:
+              parameters:
+              - name: dockerfile-to-build
+                value: "{{inputs.parameters.dockerfile-to-build}}"
+              - name: dockerfile-context
+                value: "{{inputs.parameters.dockerfile-context}}"
+              - name: dockerfile-tag
+                value: "{{inputs.parameters.dockerfile-tag}}"
+              artifacts:
+              - name: git-repo
+                from: "{{inputs.artifacts.git-repo}}"
+          - name: docker-in-docker
+            template: docker-in-docker
+            when: "{{inputs.parameters.provider}} == dockerInDocker"
+            arguments:
+              parameters:
+              - name: dockerfile-to-build
+                value: "{{inputs.parameters.dockerfile-to-build}}"
+              - name: dockerfile-context
+                value: "{{inputs.parameters.dockerfile-context}}"
+              - name: dockerfile-tag
+                value: "{{inputs.parameters.dockerfile-tag}}"
+              - name: project-shorthand
+                value: "{{inputs.parameters.project-shorthand}}"
+              artifacts:
+              - name: git-repo
+                from: "{{inputs.artifacts.git-repo}}"
+    - name: makisu-alpine
+      inputs:
+        parameters:
+        - name: dockerfile-to-build
+        - name: dockerfile-context
+        - name: dockerfile-tag
+        - name: redis-cache-address
+          value: "{{inputs.parameters.redis-cache-address}}"
+        artifacts:
+        - name: git-repo
+          path: "{{inputs.parameters.git-clone-artifact-path}}"
+      script:
+        image: gcr.io/uber-container-tools/makisu-alpine:v0.1.17
+        command: [sh]
+        workingDir: "{{inputs.artifacts.git-repo.path}}"
+        source: |
+          /makisu-internal/makisu build \
+            --registry-config=/registry-config/registries-bdata-ml.yaml \
+            --push=gcr.io \
+            --modifyfs=true \
+            --compression=no \
+            --redis-cache-addr={{inputs.parameters.redis-cache-address}} \
+            -t={{inputs.parameters.dockerfile-tag}} \
+            -f={{inputs.artifacts.git-repo.path}}/{{inputs.parameters.dockerfile-to-build}} \
+            {{inputs.parameters.dockerfile-context}}
+        volumeMounts:
+          - name: docker-registry-config
+            mountPath: /registry-config
+    - name: makisu
+      inputs:
+        parameters:
+        - name: dockerfile-to-build
+        - name: dockerfile-context
+        - name: dockerfile-tag
+        - name: redis-cache-address
+          value: "{{inputs.parameters.redis-cache-address}}"
+        artifacts:
+        - name: git-repo
+          path: "{{inputs.parameters.git-clone-artifact-path}}"
+      container:
+        image: gcr.io/uber-container-tools/makisu:latest
+        args: [
+          "build",
+          "--registry-config=/registry-config/registries-bdata-ml.yaml",
+          "--push=gcr.io",
```

Comment:
We should likely support a `registry` input param so that we get non-gcr repo support for DockerHub, ECR, Harbor, etc..

Reply:
Also internal private registries.